docs/source/remote-data-services.rst

```diff
@@ -0,0 +1,155 @@
+Remote Data Services
+====================
+
+As described in Section :doc:`Internal Data Ingestion <bytes>`, various
+user-facing functions such as ``dataframe.read_parquet`` and lower level
+byte-manipulating functions may point to data
+that lives not on the local storage of the workers, but on a remote system
+such as Amazon S3.
+
+In this section we describe how to use the various known back-end storage
+systems. Below we give some details for interested developers on how further
+storage back-ends may de provided for dask's use.
+
+
+Known Storage Implementations
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+When specifying a storage location, a URL should be provided using the geneal
+form ``protocol://path/to/data``.
+If no protocol is provided, the local file-system is assumed (same as
+``file://``). Two methods exist for passing parameters to the backend
+file-system driver: extending the URL to include username, password, server,
+port, etc.; and providing ``storage_options``, a dictionary of parameters
+to pass on.
+Further details on how to provide configuration for each backend
+is listed next.
+
+Each back-end has additional installation requirements and may not be
+available at runtime. The dictionary ``dask.bytes.core._filesystems``
+contains the currently available file-systems. Some require
+appropriate imports before use.
+
+The following list gives the protocol shorthands and the back-ends they refer
+to:
+
+    - ``file:`` - the local file system, default in the absence of any protocol
+
+    - ``hdfs:`` - Hadoop Distributed File System, a for resilient, replicated
+      files within a cluster, using the library hdfs3_
+
+    - ``s3:`` - Amazon S3 remote binary store, often used with Amazon EC2,
+      using the library s3fs_
+
+    .. - # ``adlfs:`` - Azure Data-lake cloud storage, for use with the Microsoft
+      Azure platform, using azure-data-lake-store-python_
+
+    .. - ``gcsfs:`` - Google Cloud Storage, typically used with Google Compute
+      resource, using gcsfs_
```

Comment:
Azure is not actually available, because there is no dask-azure class yet (or mapping), but it would be super-trivial to make one. Unfortunately, I don't have any access to Azure test it out now, but I could ask. Also, it's not clear where this code could live, I'm pretty sure they wouldn't want it in their repo.